server/src/main/java/org/elasticsearch/inference/UnifiedCompletionRequest.java

```diff
@@ -122,7 +122,12 @@ public record Message(Content content, String role, @Nullable String name, @Null
         );
 
         static {
-            PARSER.declareField(constructorArg(), (p, c) -> parseContent(p), new ParseField("content"), ObjectParser.ValueType.VALUE_ARRAY);
+            PARSER.declareField(
+                optionalConstructorArg(),
```

Comment:
I think we need to read and write the content field in the stream as optional. So in the `Message` constructor and in the `writeTo` I think we switch it to `in.readOptionalNamedWriteable(Content.class)`

Reply:
good catch!